### PR TITLE
Feature/ST 24 week forecast

### PR DESCRIPTION
**Pull request by heidebloome**
--> 3 day forecast 

--> fetch data
--> showing data in template
--> desktop and mobile layouts of 3rd page
--> dark mode for 3rd page

### CODE REVIEW

File: weather-app/src/app/week-forecast/week-forecast.component.ts [hunk 1]
```diff
@@ -1,97 +1,83 @@
...
-  sevenColumnChecked = false;
+export class WeekForecastComponent implements OnInit {
+  public searchQuery: string = '';
+  public forecastData: Observable<Object> = new Observable();
+  public currentCity: string = '';
+  public currentCountry: string = '';
+  public forecast!: any;
+
+  private indexArr = [10, 14, 18];
+  private weekday = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
```
Comment (chifa) [inline weather-app/src/app/week-forecast/week-forecast.component.ts:21]: I'd suggest to move it to an enum. weekday.enum.ts or something like that

File: weather-app/src/app/week-forecast/week-forecast.component.ts [hunk 2]
```diff
@@ -1,97 +1,83 @@
...
+  private month = ['01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11', '12'];
```
Comment (chifa) [inline weather-app/src/app/week-forecast/week-forecast.component.ts:22]: Same for month numbers and names

File: weather-app/src/app/week-forecast/week-forecast.component.ts [hunk 3]
```diff
@@ -1,97 +1,83 @@
...
-  sevenColumn = true;
-  sevenColumnChecked = false;
+export class WeekForecastComponent implements OnInit {
+  public searchQuery: string = '';
+  public forecastData: Observable<Object> = new Observable();
+  public currentCity: string = '';
+  public currentCountry: string = '';
+  public forecast!: any;
+
+  private indexArr = [10, 14, 18];
```
Comment (chifa) [inline weather-app/src/app/week-forecast/week-forecast.component.ts:20]: `indexArr` is a little misleading name, no idea why it keeps 10, 14 and 18 "magic numbers"

File: weather-app/src/app/week-forecast/week-forecast.component.ts [hunk 4]
```diff
@@ -1,97 +1,83 @@
...
+          if (this.indexArr.includes(index)) {
+           return item;
+          }
+        })
+        return day = day.hour;
+      })
+    })
+  }
+
+  parseNumber(num: any): number {
```
[the remark this answers is not in the document]
Comment (heidebloome) [reply weather-app/src/app/week-forecast/week-forecast.component.ts:62]: done

File: weather-app/src/app/week-forecast/week-forecast.component.ts (hunk 1 above)
Comment (heidebloome) [reply weather-app/src/app/week-forecast/week-forecast.component.ts:21]: done

File: weather-app/src/app/week-forecast/week-forecast.component.ts (hunk 2 above)
Comment (heidebloome) [reply weather-app/src/app/week-forecast/week-forecast.component.ts:22]: done

File: weather-app/src/app/week-forecast/week-forecast.component.ts (hunk 3 above)
Comment (heidebloome) [reply weather-app/src/app/week-forecast/week-forecast.component.ts:20]: done

**Review by chifa — APPROVED** [review]

**Pull request merged**